Makefile

```diff
@@ -106,6 +106,31 @@ test-tvos:
 	fi; \
 	xcodebuild test -workspace $(PROJECT_NAME).xcworkspace -scheme "$$final_scheme" -destination 'platform=tvOS Simulator,name=Apple TV' -derivedDataPath build/out -resultBundlePath tvosresults.xcresult -enableCodeCoverage YES ADB_SKIP_LINT=YES
 
+# Runs the Edge Network (Konductor) integration tests after installing pod dependencies
+# Usage: 
+# make test-integration-upstream EDGE_ENVIRONMENT=<environment> EDGE_LOCATION_HINT=<location_hint>
+# If EDGE_ENVIRONMENT is not specified, test target will use its default value.
+.SILENT: test-integration-upstream # Silences Makefile's automatic echo of commands
+test-integration-upstream: pod-install; \
+	if [ -z "$$EDGE_ENVIRONMENT" ]; then \
+		echo ''; \
+		echo '-------------------------- WARNING -------------------------------'; \
+		echo 'EDGE_ENVIRONMENT was NOT set; the test will use its default value.'; \
+		echo '------------------------------------------------------------------'; \
+		echo ''; \
+	fi; \
+	xcodebuild test \
+	-quiet \
+	-workspace $(PROJECT_NAME).xcworkspace \
+	-scheme UpstreamIntegrationTests \
+	-destination 'platform=iOS Simulator,name=iPhone 14' \
+	-derivedDataPath build/out \
+	-resultBundlePath iosresults.xcresult \
```

Comment:
We should add folder to the coverage to avoid conflicts with coverage from our unit and functional tests
`-resultBundlePath coverage/upstreamIntegrationTest/iosresults.xcresult`

Reply:
nit: command name `test-integration-upstream-ios`

@emdobrin does it make sense to have it with tvOS as well?

Reply:
In my opinion we should thoroughly test the various platforms at functional test level and keep this simple for now, we can revisit this if we have platform specific test cases in the future.